packages/x-charts/src/ChartsXAxis/ChartsGroupedXAxis.tsx

```diff
@@ -0,0 +1,244 @@
+'use client';
+import * as React from 'react';
+import useSlotProps from '@mui/utils/useSlotProps';
+import { useThemeProps, useTheme } from '@mui/material/styles';
+import { useRtl } from '@mui/system/RtlProvider';
+import { ChartsXAxisProps, type AxisConfig, type AxisGroupingConfig } from '../models/axis';
+import { ChartsText, ChartsTextProps } from '../ChartsText';
+import { useDrawingArea } from '../hooks/useDrawingArea';
+import { isInfinity } from '../internals/isInfinity';
+import { isBandScale } from '../internals/isBandScale';
+import { useChartContext } from '../context/ChartProvider/useChartContext';
+import { useXAxes } from '../hooks/useAxis';
+import { getDefaultBaseline, getDefaultTextAnchor } from '../ChartsText/defaultTextPlacement';
+import { invertTextAnchor } from '../internals/invertTextAnchor';
+import { defaultProps, TICK_LABEL_GAP, XAxisRoot, useUtilityClasses } from './utilities';
+import { useTicksGrouped } from '../hooks/useTicksGrouped';
+
+const DEFAULT_GROUPING_CONFIG = {
+  tickSize: 6,
+};
+
+const calculateTickSize = (
+  groupingConfig: AxisGroupingConfig | undefined,
+  groupIndex: number,
+  tickSize: number | undefined,
+  isConfigArray: boolean = false,
+): number => {
+  // If the groupingConfig is an array we expect the user to provide a `tickSize` for each group.
+  if (isConfigArray) {
+    return groupingConfig?.tickSize ?? DEFAULT_GROUPING_CONFIG.tickSize;
+  }
+
+  // Else if it is an object, the provided `tickSize` applies to all groups incrementally.
+  const computedTickSize = groupingConfig?.tickSize ?? tickSize ?? DEFAULT_GROUPING_CONFIG.tickSize;
+
+  // If only the `tickSize` property is provided, we assume it applies to all groups.
+  // The first tick will be at `tickSize`, while every subsequent group will be
+  // multiplied by the group index times two and summed to the first tick size.
+  // This allows for a consistent spacing between groups.
+  return computedTickSize * groupIndex * 2 + computedTickSize;
+};
+
+const getGroupingConfig = (
+  groupingConfig: AxisConfig['groupingConfig'],
+  groupIndex: number,
+  tickSize: number | undefined,
+) => {
+  if (!groupingConfig) {
+    return {
+      tickSize: calculateTickSize(undefined, groupIndex, tickSize),
+    };
+  }
+
+  const isConfigArray = Array.isArray(groupingConfig);
+  const config = isConfigArray ? groupingConfig[groupIndex] : groupingConfig;
+
+  return {
+    ...DEFAULT_GROUPING_CONFIG,
+    ...config,
+    tickSize: calculateTickSize(config, groupIndex, tickSize, isConfigArray),
+  };
+};
+
+/**
+ * @ignore - internal component.
+ */
+function ChartsGroupedXAxis(inProps: ChartsXAxisProps) {
```

Comment:
I've removed a lot of checks/functions that are present in the "Single" axis, like shortenLabels and visibility checks.